### ISSUE TITLE
Doesn't get strings from theme.json, patterns etc... in themes

### ISSUE THREAD

**Issue by roborourke** [wp-blocks/make-pot#27]
If you compare the output on a block theme with some patterns in the `/patterns/` directory, and a `theme.json` with string labels for colours etc... they do not appear in the generated `.pot` file as they do when using WP CLI's `wp i18n make-pot`. Is there some setting I'm missing perhaps?

It is throwing a bunch of warnings like:

```
Failed to load schema from http://develop.svn.wordpress.org/trunk/src/wp-includes/block-i18n.json. Using fallback
```

But I'm not sure why or how to resolve that, thought I'd ask first before digging much deeper.

**Comment by erikyo** [comment]
I believe I've addressed the issue that was preventing the schema from loading correctly.

Could you please test with the latest changes and confirm if the string labels from your block theme patterns and theme.json now appear in the generated .pot file, as they do with wp i18n make-pot? Your confirmation would be greatly appreciated.

**Comment by roborourke** [comment]
Thanks for looking into it :) What's the best way to test? I'm having trouble trying to run it from the repo directly via npx or a global npm install from `master` currently.

**Comment by erikyo** [comment]
That’s strange—what kind of issue are you running into?

To try again, please uninstall the globally installed command (if you have it) by running:

```bash
npm uninstall -g @wp-blocks/make-pot
```

Then, run the command using npx instead:

```bash
npx @wp-blocks/make-pot
```
Currently, both commands don’t require any arguments. By default, the source folder is the one where you run the command, and the destination is the languages/ folder. So you just need to run the commands inside your plugin or theme folder—nothing else is needed.

This way, you’ll always use the latest version from the npm registry without needing a global installation—npx downloads and runs a temporary copy of the command.

If that still doesn’t work, please copy and paste the error message you’re seeing and let me know whether you're running it inside a plugin or a theme folder.


---
Bonus for whoever what to contribute:

If you want to run it directly from the repository (e.g. from the `master` branch), here's what I suggest:

1. Clone the repository:

   ```bash
   git clone https://github.com/wp-blocks/make-pot.git
   cd make-pot
   ```

2. Install dependencies:

   ```bash
   npm install && npm run build
   ```

3. TO run the CLI directly from the source:

   ```bash
   node ./bin/index.js
   ```

4. To use the downloaded version in another project (npm link info [here](https://docs.npmjs.com/cli/v8/commands/npm-link))

   ```bash
   npm link
   ```

This will execute the command using the local source code, allowing you to test any changes on the `master` branch or your own modifications, and run the modified version in your local folders.

Note: if you've updated the Tree-sitter dependencies to newer versions, it's a good idea to run npm rebuild. This is necessary because Tree-sitter uses the N-API C++ wrapper classes, which need to be recompiled when versions change.

**Comment by roborourke** [comment]
I think these changes aren't published yet, and thanks for the tip re. tree sitter, that was indeed the problem. Running some tests now.

**Comment by erikyo** [comment]
should be updated... check the update date below 👇
https://www.npmjs.com/package/@wp-blocks/make-pot

ps thanks for your help! 🍻

**Comment by roborourke** [comment]
ah ok, I thought in the related PR I saw 1.5.1 in the package.json so was expecting to see that

**Comment by erikyo** [comment]
This evening I think I will [complete the tests](https://github.com/wp-blocks/make-pot/pull/50/commits/5586c8500648c89bb044dc8fb7bd910c244f9fc0#diff-9882ec0ded0a230b5478996b4fb73082eb68a8af01c3e29a13363c54340bafb6) on theme.json and block.json and then I will publish it.

**Comment by erikyo** [comment]
@roborourke the new version is now available on npm, if you want to give it a try. 
https://www.npmjs.com/package/@wp-blocks/make-pot

**Comment by roborourke** [comment]
Hi @erikyo sorry for the delay. I tried with this version but it keeps complaining about a missing author email with a `style.css` header like the following. I added the email address in but it didn't resolve it.

```css
/**
 * Theme Name:  HM Theme
 * Theme URI:   https://humanmade.com
 * Description: HM Theme
 * Author:      Human Made Limited <engineering@humanmade.com>
 * Author URI:  https://humanmade.com/
 * Version:     2.0.0
 * Text Domain: hm
 * Domain Path: /languages/
 *
 * License:     GPLv2+
 * License URI: http://www.gnu.org/licenses/gpl-2.0.html
 */
```

**Comment by erikyo** [comment]
Thank you and don't worry, I thank you because every feedback helps me to improve! 

I think the problem is because you don't have in your package.json your email in the author field.

try writing it like this
```
    "author": "codekraft <erik@codekraft.it>",
```

**Comment by roborourke** [comment]
Ahh ok, in this instance the theme has no package.json because that’s at the project root level. It’s a bit of an edge case, I can provide the header via the command line if I’m reading it correctly

**Comment by erikyo** [comment]
@roborourke you are correct, but I the 1.5.1 version will not work as expected for you. Let me make a quick fix and consolidate the CLI command in the same way expected by wp-cli i18n using the --header argument. At the moment it is handled but not in the correct way, sorry for the inconvenience, and thank you again for the help!

```
[--headers=<headers>]
	Array in JSON format of custom headers which will be added to the POT file. Defaults to empty array.
```

**Comment by erikyo** [comment]
@roborourke i just published v1.6.0 that should work also for your case

Add the following argument to the CLI command `--headers=email:your@email.com` and it should be fine. Thanks again for having reported the issue to me, and if by chance there are any problems, feel free to reopen the issue.
